File: frontend/src/types/bourse.ts
export interface Position {
  nom: string;
  ticker: string;
  valeur: number;
  poids: number;
  ytd: number;
  pvEur: number;
  sparkData?: number[] | null;
}

export interface Envelope {
  nom: string;
  total: number;
  ytdPct: number;
  plafond: number | null;
  versementRestant: number | null;
  positions: Position[];
}

export interface Crypto {
  nom: string;
  symbol: string;
  valeur: number;
  poids: number;
  ytd: number;
}

export interface BourseData {
  actifsFinanciers: number;
  pea: Envelope;
  cto: Envelope;
  peaPme: Envelope;
  cryptoTotal: number;
  cryptoYtdPct: number;
  cryptos: Crypto[];
}

File: backend/app/models/bourse.py
from app.models.common import CamelModel


class Position(CamelModel):
    nom: str
    ticker: str
    valeur: int
    poids: float
    ytd: float
    pv_eur: int
    spark_data: list[float] | None = None


class Envelope(CamelModel):
    nom: str
    total: int
    ytd_pct: float
    plafond: int | None = None
    versement_restant: int | None = None
    positions: list[Position]


class Crypto(CamelModel):
    nom: str
    symbol: str
    valeur: int
    poids: float
    ytd: float


class BourseData(CamelModel):
    actifs_financiers: int
    pea: Envelope
    cto: Envelope
    pea_pme: Envelope
    crypto_total: int
    crypto_ytd_pct: float
    cryptos: list[Crypto]
